graphdatascience/session/session_v2_endpoints.py

```diff
@@ -0,0 +1,35 @@
+from typing import Optional
+
+from graphdatascience.arrow_client.authenticated_flight_client import AuthenticatedArrowClient
+from graphdatascience.arrow_client.v2.write_back_client import WriteBackClient
+from graphdatascience.procedure_surface.api.catalog_endpoints import CatalogEndpoints
+from graphdatascience.procedure_surface.api.wcc_endpoints import WccEndpoints
+from graphdatascience.procedure_surface.arrow.catalog_arrow_endpoints import CatalogArrowEndpoints
+from graphdatascience.procedure_surface.arrow.wcc_arrow_endpoints import WccArrowEndpoints
+from graphdatascience.query_runner.query_runner import QueryRunner
+
+
+class SessionV2Endpoints:
+    def __init__(
+        self,
+        arrow_client: AuthenticatedArrowClient,
+        bolt_client: QueryRunner,
+        db_client: Optional[QueryRunner] = None,
+    ):
+        self._arrow_client = arrow_client
+        self._bolt_client = bolt_client
+        self._db_client = db_client
+
+        self._write_back_client = WriteBackClient(arrow_client, db_client) if db_client is not None else None
+
+    @property
+    def wcc(self) -> WccEndpoints:
+        return WccArrowEndpoints(self._arrow_client, self._write_back_client)
+
+    # sessions endpoints: clear if arrow or cypher. for plugin switch between cypher and arrow based on boolean (gds constructor)
+
+    # TODO: endpoints sharing a common suffix such as graph catalog and graph sample options need to have a shared interface to return here
+
+    @property
+    def graph(self) -> CatalogEndpoints:
+        return CatalogArrowEndpoints(self._arrow_client, self._db_client)
```

Comment:
AFAIK - project is not included here yet?

Reply:
it differs per implementation. but that will make it impossible to include here.

I think for these SessionEndpoints we can always return the ArrowEndpoints .. but these wouldnt have the doc strings?
makes me lean towards adding the remote project endpoint to the general api class 🤔

Reply:
signature overload could be an idea for this case? -- however we dont need the non remote one for the Session case. And for plugin we dont need the remote one.

could have different graph catalog interfaces for these two to return to the user